x-pack/plugins/secops/server/lib/suricata/types.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SourceConfiguration, SuricataEvents, TimerangeInput } from '../../../common/graphql/types';
+import { JsonObject } from '../../../common/typed_json';
+import { FrameworkRequest } from '../framework';
+
+export interface SuricataAdapter {
+  getEvents(req: FrameworkRequest, options: SuricataRequestOptions): Promise<SuricataEvents[]>;
+}
+
+export type ESQuery = ESRangeQuery | ESQueryStringQuery | ESMatchQuery | JsonObject;
+
+export interface ESRangeQuery {
+  range: {
+    [name: string]: {
+      gte: number;
+      lte: number;
+      format: string;
+    };
+  };
+}
+
+export interface ESMatchQuery {
+  match: {
+    [name: string]: {
+      query: string;
```

Comment:
Can this interface represent the full set of `match` queries that can be expressed? For example, the [Match Query API Docs page](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html) provides the following example, where it queries a field named `message` via `query` (as modeled in this `interface`), but in addition to `query`, the example also uses `operator ` and `zero_terms_query`:

```
curl -X GET "localhost:9200/_search" -H 'Content-Type: application/json' -d'
{
    "query": {
        "match" : {
            "message" : {
                "query" : "to be or not to be",
                "operator" : "and",
                "zero_terms_query": "all"
            }
        }
    }
}
'
``` 

The [DefinitelyTyped mappings](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/elasticsearch/index.d.ts) mappings for [elasticsearch.js](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/about.html) may help answer questions like this. See also [bodybuilder.js](https://bodybuilder.js.org/)

Reply:
👍